### ISSUE TITLE
[JENKINS-23524] Build Pipeline does not discover downstream promoted jobs?

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/build-pipeline-plugin#268]
<p>I installed Build Pipeline and tried it on my jobs. They work by invoking a build as a promotion action. I was disappointed that the pipeline view failed to follow my builds except where there was a direct chain of automatic chained jobs.<br/>
I guess perhaps my expectation doesn't match the pipeline concept? Or perhaps my build flow is not a pipeline?<br/>
Basically, we have a continuous build job which allows a selected build to be promoted. The promotion action causes another build to be executed. Generally we do not have an automatic build chain, but a manual promotion chain. This approach works better for us, since we don't have to maintain (and archive) any build history unless a build is promoted.<br/>
Perhaps my expectation is in error, however I think it would be a great enhancement to a great plugin!</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mwigzell">mwigzell</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23524" target="_blank">Build Pipeline does not discover downstream promoted jobs?</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: build-pipeline-plugin
<li><b>label(s)</b>: promoted-build-chain
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 20260107-091649
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I installed Build Pipeline and tried it on my jobs. They work by invoking a build as a promotion action. I was disappointed that the pipeline view failed to follow my builds except where there was a direct chain of automatic chained jobs.
I guess perhaps my expectation doesn't match the pipeline concept? Or perhaps my build flow is not a pipeline?
Basically, we have a continuous build job which allows a selected build to be promoted. The promotion action causes another build to be executed. Generally we do not have an automatic build chain, but a manual promotion chain. This approach works better for us, since we don't have to maintain (and archive) any build history unless a build is promoted.
Perhaps my expectation is in error, however I think it would be a great enhancement to a great plugin!</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins 1.554, Build Pipeline version 1.4.3</code></li></ul>
